### DISCUSSION TITLE
Don't select empty zone

### DISCUSSION THREAD

**Discussion by pureby** [sylikc/jpegview#169] [Ideas]
1. It would be better if selection started within the image boundaries and not select empty space around the image (regardless whether the selection is initiated from inside or outside the image):

<img src="https://user-images.githubusercontent.com/6698649/225102597-ca42fd18-73d8-4ac5-a16c-16e5d1f7c11c.jpg" width=25% height=25%>

**_[24/12/2023 EDIT]_ A couple new points for improving the selection mechanism:**

2. **[BUG] Selection resizing at Bottom-Left and Top-Right corners incorrect behavior.**
After the selection rectangle is drawn on the image, these corners do not behave properly when dragged. The "anchor points" for these corners should be **the opposite corners** (by "anchor points" I mean the corners to which selection should be "locked", and not move while being resized). Right now wrong "anchor points" are chosen, making selection resizing behave unintuitive and somewhat unpredictable when dragging.
_- This only happens when using a set **selection ratio** — selection in **freeform mode** behaves correctly._
_- Top-Left and Bottom-Right corners also behave correctly._  

3. **Disable automatic menu pop-up after selection.**
`SelectionAutoPopup=false` setting for the .ini file would be ideal (use **Right Mouse Button** to open menu instead).
This menu is often distracting when cropping because it covers a big area of the image, and you don't always know whether you got the selection just right on the first try. Or the next try.
Right now, to hide the menu user must either: sniper-aim and click the selection rectangle (which may unintentionally resize the selection), OR go to `Crop mode >`, then find and select the right option (time-consuming) , OR press the `Alt` key (which re-initiates the selection).
_**Alternatively,** if the above is difficult to implement for some reason, add a `Close this menu` item to the aforementioned menu, so it can be easily hidden without sniper-precise aiming._

**Comment by sylikc** [comment]
There was a previous feature request ... #144 request number 2.  I am not quite sure... I need to look into it.

When I was doing the crop AR stuff it, I briefly looked into it, but it didn't look straightforward at the time

**Reply by pureby** [reply]
Thanks for the clarification! 

Sorry I overlooked the existing suggestion, I didn't think about searching "issues" for stuff like this.

**Reply by sylikc** [reply]
that's alright.  it would've been hard to find haha.  I just know them by heart since I read through them and take notes 😎

**Comment by pureby** [comment]
Hi @sylikc, in case improving selection is something you're interested in for a future release, I have added a couple new points of improvement:

2. [BUG] Selection resizing at **Bottom-Left** and **Top-Right** corners incorrect behavior.
3. `SelectionAutoPopup=false` / `Close this menu` option.

See the OP for detailed description.
